# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build libsigsegv-build readline-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.69; am-1.15; lt-2.4.6)
aclocal -I m4; autoheader; autoconf; automake -a -c
cd extension; libtoolize -c -f; aclocal -I ../m4; autoheader; autoconf; automake -a -c; cd -
./configure --prefix=

# build
make

# install
install -d $DESTDIR/bin
install gawk $DESTDIR/bin
ln -sf gawk $DESTDIR/bin/awk
install -d $DESTDIR/lib/gawk
install extension/.libs/inplace.so $DESTDIR/lib/gawk
install -d $DESTDIR/share/awk
install -m 644 awklib/eg/lib/inplace.awk $DESTDIR/share/awk

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/gawk-build.pc << EOF
Name: gawk
Version: 4.2.1
Description: GNU awk
Requires: $REQUIRES

exec=\\
/bin/awk \\
/bin/gawk \\
/lib/gawk/inplace.so \\
/share/awk/inplace.awk
EOF

exit

# patch
--- gawk-4.2.1.orig/configure.ac
+++ gawk-4.2.1/configure.ac
@@ -42,6 +42,8 @@
 AC_PREREQ([2.69])
 AM_INIT_AUTOMAKE([1.15 dist-xz dist-lzip])
 
+AM_MAINTAINER_MODE
+
 AC_CONFIG_MACRO_DIR([m4])
 
 dnl Additional argument stuff
--- gawk-4.2.1.orig/extension/configure.ac
+++ gawk-4.2.1/extension/configure.ac
@@ -32,6 +32,7 @@
 
 AM_INIT_AUTOMAKE([1.15 -Wall -Werror])
 
+AM_MAINTAINER_MODE
 
 AC_USE_SYSTEM_EXTENSIONS
 AC_ZOS_USS

# source
https://ftp.gnu.org/gnu/gawk/gawk-4.2.1.tar.gz
